report.go

```diff
@@ -1,106 +1,157 @@
-func FindMatches(baseURL string) []string {
-    resp, _ := soup.Get(baseURL)
-
-    doc := soup.HTMLParse(resp)
-    anchors := doc.FindAll("a")
-    exist := make(map[string]bool)
-    var urls []string
-    for _, anchor := range anchors {
-        url := anchor.Attrs()["href"]
-        if strings.Contains(url, "match") && !exist[url] {
-            exist[url] = true
-            urls = append(urls, url)
-        }
-    }
-    return urls
+// FindLinks find all analysis related hrefs in HTML
+func findLinks(baseURL string) []string {
+	resp, _ := soup.Get(baseURL)
+
+	doc := soup.HTMLParse(resp)
+	anchors := doc.FindAll("a")
+	exist := make(map[string]bool)
+	var urls []string
+	for _, anchor := range anchors {
+		url := anchor.Attrs()["href"]
+		if strings.Contains(url, "match") && !exist[url] {
+			exist[url] = true
+			urls = append(urls, url)
+		}
+	}
+	return urls
 }
 
-// DownloadURL into files
+// DownloadURL downloads URL to file
 func DownloadURL(url string, filename string) error {
-    resp, err := soup.Get(url)
-    if err != nil {
-        return err
-    }
+	resp, err := soup.Get(url)
+	if err != nil {
+		return err
+	}
 
-    f, err := os.Create(filename)
-    defer f.Close()
-    if err != nil {
-        return err
-    }
+	f, err := os.Create(filename)
+	defer f.Close()
+	if err != nil {
+		return err
+	}
 
-    reg := regexp.MustCompile(`http:\/\/moss.stanford.edu\/results\/\d*\/`)
-    resp = reg.ReplaceAllString(resp, "")
+	reg := regexp.MustCompile(`http:\/\/moss.stanford.edu\/results\/\d*\/`)
+	resp = reg.ReplaceAllString(resp, "")
 
-    f.Write([]byte(resp))
+	_, err = f.Write([]byte(resp))
+	if err != nil {
+		return err
+	}
 
-    return nil
+	return nil
 }
 
-// DownloadMoss : Download all HTMLs from given moss URL recursively
-func DownloadMoss(baseURL string) error {
-    basedir := "moss.stanford.edu"
-    err := os.MkdirAll(basedir, 0755)
-    if err != nil {
-        panic("Can't create a directory")
-    }
-
-    urls := FindMatches(baseURL)
-    DownloadURL(baseURL + "/index.html", basedir + "/index.html")
-    for index := range urls {
-        DownloadURL(fmt.Sprintf("%v/match%v.html", baseURL, index), fmt.Sprintf("%v/match%v.html", basedir, index))
-        DownloadURL(fmt.Sprintf("%v/match%v-top.html", baseURL, index), fmt.Sprintf("%v/match%v-top.html", basedir, index))
-        DownloadURL(fmt.Sprintf("%v/match%v-0.html", baseURL, index), fmt.Sprintf("%v/match%v-0.html", basedir, index))
-        DownloadURL(fmt.Sprintf("%v/match%v-1.html", baseURL, index), fmt.Sprintf("%v/match%v-1.html", basedir, index))
-    }
-    return nil
+// Download downloads all HTML files from given URL recursively
+func Download(baseURL string) error {
+	basedir := "moss.stanford.edu"
+	err := os.MkdirAll(basedir, 0755)
+	if err != nil {
+		return fmt.Errorf("Can't create a directory")
+	}
+
+	urls := findLinks(baseURL)
+
+	DownloadURL(baseURL+"/index.html", basedir+"/index.html")
```

Comment:
Can you use [`URL.Parse`](https://golang.org/pkg/net/url/#URL.Parse) instead of concatenation across the change?